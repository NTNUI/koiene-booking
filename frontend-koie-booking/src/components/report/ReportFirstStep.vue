<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.reportTitle') }}</h1>
    <v-layout :class="$style.separator">
      <v-layout class="px-4">
        <p>{{ $t('report.reportDescription') }}</p>
      </v-layout>
      <v-layout class="px-4">
        <p>
          {{ $t('report.reportYourTripTo') }}
          <span :class="$style.capitalize">{{ koie }}</span>
          {{ $t('report.reportFrom') }}
          {{ fromDate }}
          {{ $t('report.reportTo') }}
          {{ toDate }}.
        </p>
      </v-layout>
    </v-layout>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { formatDate } from '@/utils/dates';

export default Vue.extend({
  name: 'ReportFirstStep',
  computed: {
    koie(): string {
      return this.$store.state.report.bookingInfo.koie;
    },
    fromDate(): string {
      return formatDate(this.$store.state.report.bookingInfo.from_date, 'DD.MM.YYYY');
    },
    toDate(): string {
      return formatDate(this.$store.state.report.bookingInfo.to_date, 'DD.MM.YYYY');
    },
  },
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
.capitalize {
  text-transform: capitalize;
}
</style>
