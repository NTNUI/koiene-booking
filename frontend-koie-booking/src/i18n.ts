import Vue from 'vue';
import VueI18n, { LocaleMessages } from 'vue-i18n';
import enJson from './locales/en.json';

Vue.use(VueI18n);

function loadLocaleMessages(): LocaleMessages {
  // Testing currently only supports manually importing of json-files. English is the only language that gets loaded now
  let locales: any;
  const messages: LocaleMessages = {};
  if (process.env.NODE_ENV === 'test') {
    messages['en'] = enJson;
  } else {
    locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
    locales.keys().forEach((key: any) => {
      const matched = key.match(/([A-Za-z0-9-_]+)\./i);
      if (matched && matched.length > 1) {
        const locale = matched[1];
        messages[locale] = locales(key);
      }
    });
  }
  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages()
});
