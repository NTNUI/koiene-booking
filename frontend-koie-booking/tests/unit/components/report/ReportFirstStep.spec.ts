import Vue from 'vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import i18n from '@/i18n';

Vue.use(Vuetify);

// Utilities
import { mount, createLocalVue } from '@vue/test-utils';

// Components or views
import ReportFirstStep from '@/components/report/ReportFirstStep.vue';

describe('Component ReportFirstStep.vue', () => {
  let wrapper: any;
  let localVue: any;
  let vuetify: any;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    localVue.use(Vuetify);
    localVue.use(Vuex);
  });

  it('Matches snapshot', async () => {
    wrapper = mount(ReportFirstStep, {
      localVue,
      vuetify,
      i18n,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
