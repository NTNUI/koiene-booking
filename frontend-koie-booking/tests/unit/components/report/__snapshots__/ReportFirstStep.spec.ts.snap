// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ReportFirstStep.vue Matches snapshot 1`] = `
<div class="layout dark container">
  <h1 class="heading">Digital Cabin Report</h1>
  <div class="layout separator">
    <div class="layout px-4">
      <p>The cabin group is run purely on volunteer work and we rely on your feedback to be able to run the cabins.</p>
    </div>
  </div>
</div>
`;
